Port more identifiers from LegacyNullableObjectIdentifier to ObjectIdentifier

cdumez opened the pull request:
#### 34fca85581d4fb6c6515240b38e725b1cf80de38
<pre>
Port more identifiers from LegacyNullableObjectIdentifier to ObjectIdentifier
<a href="https://bugs.webkit.org/show_bug.cgi?id=278976">https://bugs.webkit.org/show_bug.cgi?id=278976</a>

Reviewed by Youenn Fablet.

Port more identifiers from LegacyNullableObjectIdentifier to ObjectIdentifier.

Also drop a couple of MarkSurfacesAsVolatileRequestIdentifier data members which would
never get initialized.

* Source/WebKit/Shared/Authentication/cocoa/AuthenticationManagerCocoa.mm:
(WebKit::AuthenticationManager::initializeConnection):
* Source/WebKit/Shared/IdentifierTypes.h:
* Source/WebKit/Shared/MarkSurfacesAsVolatileRequestIdentifier.h:
* Source/WebKit/Shared/UserContentControllerIdentifier.h:
* Source/WebKit/Shared/WTFArgumentCoders.serialization.in:
* Source/WebKit/UIProcess/Authentication/AuthenticationChallengeProxy.cpp:
* Source/WebKit/UIProcess/GeolocationIdentifier.h:
* Source/WebKit/UIProcess/PDF/WKPDFHUDView.mm:
(-[WKPDFHUDView _performActionForControl:]):
* Source/WebKit/UIProcess/UserContent/WebUserContentControllerProxy.cpp:
(WebKit::WebUserContentControllerProxy::parameters const):
* Source/WebKit/UIProcess/WebPageGroup.cpp:
(WebKit::pageGroupData):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::creationParameters):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/WebProcess/GPU/graphics/RemoteImageBufferSetProxy.cpp:
(WebKit::RemoteImageBufferSetProxy::setConfirmedVolatility):
(WebKit::RemoteImageBufferSetProxy::clearVolatility):
(WebKit::RemoteImageBufferSetProxy::clearVolatilityUntilAfter): Deleted.
* Source/WebKit/WebProcess/GPU/graphics/RemoteImageBufferSetProxy.h:
* Source/WebKit/WebProcess/GPU/graphics/RemoteRenderingBackendProxy.cpp:
(WebKit::RemoteRenderingBackendProxy::prepareImageBufferSetsForDisplay):
(WebKit::RemoteRenderingBackendProxy::didMarkLayersAsVolatile):
* Source/WebKit/WebProcess/GPU/graphics/RemoteRenderingBackendProxy.h:
* Source/WebKit/WebProcess/Geolocation/GeolocationPermissionRequestManager.cpp:
(WebKit::GeolocationPermissionRequestManager::cancelRequestForGeolocation):
* Source/WebKit/WebProcess/Plugins/PDFPluginIdentifier.h:

Canonical link: <a href="https://commits.webkit.org/283093@main">https://commits.webkit.org/283093@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/94a62eea5627adc0ec321917b933ad2a63d1390b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/65106 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/44473 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/17720 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/69130 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/15712 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/52256 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/15994 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/52302 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/10857 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/68172 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/41118 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/56354 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/32942 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/37788 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/14588 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/59695 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/14064 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/70835 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/9058 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/13547 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/59630 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/9090 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/56414 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/59886 "Found 1 new API test failure: /WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/text/selections (failure)") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/7481 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/1158 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/9885 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/40285 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/41362 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/42543 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/41106 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

RupinMittal commented on Source/WebKit/UIProcess/WebPageGroup.cpp:
```diff
@@ -55,22 +55,26 @@ Ref<WebPageGroup> WebPageGroup::create(const String& identifier)
+        validIdentifier = identifier;
     else
-        data.identifier = makeString("__uniquePageGroupID-"_s, data.pageGroupID.toUInt64());
+        validIdentifier = makeString("__uniquePageGroupID-"_s, pageGroupID.toUInt64());
```
I saw in `ObjectIdentifier.h` that `toUInt64()` calls `toRawValue()` and there's a comment that says "Use toRawValue instead". I wanted to ask if we prefer directly using `toRawValue()` in cases like this?

cdumez replied:
I think it is wishful thinking at this point. Most call sites use toUint64() and ObjectIdentifier is an integer 99.9% of the time. If someone really wants to drop toUint64(), then they’ll just to a mass rename.

RupinMittal replied:
Ok got it, that makes sense.

youennf commented on Source/WebKit/UIProcess/UserContent/WebUserContentControllerProxy.cpp:
```diff
@@ -110,34 +110,49 @@ void WebUserContentControllerProxy::addProcess(WebProcessProxy& webProcessProxy)
         webProcessProxy.addMessageReceiver(Messages::WebUserContentControllerProxy::messageReceiverName(), identifier(), *this);
 }
 
+#if 0
```
Why is it there?

cdumez replied:
Oops, will remove.

youennf commented on Source/WebKit/UIProcess/UserContent/WebUserContentControllerProxy.cpp:
```diff
@@ -110,34 +110,49 @@ void WebUserContentControllerProxy::addProcess(WebProcessProxy& webProcessProxy)
 
+    return {
+        identifier()
+        , WTFMove(userContentWorlds)
```
```suggestion
        WTFMove(userContentWorlds),
```

cdumez replied:
Your proposal doesn't play well with the `#if ENABLE(CONTENT_EXTENSIONS)` below.

youennf commented on Source/WebKit/UIProcess/WebPageProxy.cpp:
```diff
@@ -10539,7 +10539,15 @@ bool WebPageProxy::useGPUProcessForDOMRenderingEnabled() const
 WebPageCreationParameters WebPageProxy::creationParameters(WebProcessProxy& process, DrawingAreaProxy& drawingArea, WebCore::FrameIdentifier mainFrameIdentifier, std::optional<RemotePageParameters>&& remotePageParameters, bool isProcessSwap, RefPtr<API::WebsitePolicies>&& websitePolicies)
 {
-    WebPageCreationParameters parameters;
+    std::reference_wrapper<WebUserContentControllerProxy> userContentController(m_userContentController.get());
```
This seems a bit odd, why not using a Ref? creationParameters is probably not called that often anyway.

cdumez replied:
I merely moved the code but I'll look into making this change.

youennf commented on Source/WebKit/UIProcess/PDF/WKPDFHUDView.mm:
```diff
@@ -254,13 +254,13 @@ - (void)_performActionForControl:(NSString *)control
         return;
     if ([control isEqualToString:PDFHUDZoomInControl])
-        page->pdfZoomIn(_pluginIdentifier);
+        page->pdfZoomIn(*_pluginIdentifier);
```
How do we know `!!_pluginIdentifier`?

cdumez replied:
Because it is set unconditionally in `initWithFrame`.  With Objective C, I didn't find a way to avoid using Markable even though this data member is always initialized in the "constructor".

cdumez committed:
Port more identifiers from LegacyNullableObjectIdentifier to ObjectIdentifier
https://bugs.webkit.org/show_bug.cgi?id=278976

Reviewed by Youenn Fablet.

Port more identifiers from LegacyNullableObjectIdentifier to ObjectIdentifier.

Also drop a couple of MarkSurfacesAsVolatileRequestIdentifier data members which would
never get initialized.

* Source/WebKit/Shared/Authentication/cocoa/AuthenticationManagerCocoa.mm:
(WebKit::AuthenticationManager::initializeConnection):
* Source/WebKit/Shared/IdentifierTypes.h:
* Source/WebKit/Shared/MarkSurfacesAsVolatileRequestIdentifier.h:
* Source/WebKit/Shared/UserContentControllerIdentifier.h:
* Source/WebKit/Shared/WTFArgumentCoders.serialization.in:
* Source/WebKit/UIProcess/Authentication/AuthenticationChallengeProxy.cpp:
* Source/WebKit/UIProcess/GeolocationIdentifier.h:
* Source/WebKit/UIProcess/PDF/WKPDFHUDView.mm:
(-[WKPDFHUDView _performActionForControl:]):
* Source/WebKit/UIProcess/UserContent/WebUserContentControllerProxy.cpp:
(WebKit::WebUserContentControllerProxy::parameters const):
* Source/WebKit/UIProcess/WebPageGroup.cpp:
(WebKit::pageGroupData):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::creationParameters):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/WebProcess/GPU/graphics/RemoteImageBufferSetProxy.cpp:
(WebKit::RemoteImageBufferSetProxy::setConfirmedVolatility):
(WebKit::RemoteImageBufferSetProxy::clearVolatility):
(WebKit::RemoteImageBufferSetProxy::clearVolatilityUntilAfter): Deleted.
* Source/WebKit/WebProcess/GPU/graphics/RemoteImageBufferSetProxy.h:
* Source/WebKit/WebProcess/GPU/graphics/RemoteRenderingBackendProxy.cpp:
(WebKit::RemoteRenderingBackendProxy::prepareImageBufferSetsForDisplay):
(WebKit::RemoteRenderingBackendProxy::didMarkLayersAsVolatile):
* Source/WebKit/WebProcess/GPU/graphics/RemoteRenderingBackendProxy.h:
* Source/WebKit/WebProcess/Geolocation/GeolocationPermissionRequestManager.cpp:
(WebKit::GeolocationPermissionRequestManager::cancelRequestForGeolocation):
* Source/WebKit/WebProcess/Plugins/PDFPluginIdentifier.h:

Canonical link: https://commits.webkit.org/283093@main
